pvlib/iotools/solaranywhere.py

```diff
@@ -0,0 +1,297 @@
+        start = pd.to_datetime(start)
+        end = pd.to_datetime(end)
+        # start/end are required to have an associated time zone
+        if start.tz is None:
+            start = start.tz_localize('UTC')
+        if end.tz is None:
+            end = end.tz_localize('UTC')
+        payload['Options']["StartTime"] = start.isoformat()
+        payload['Options']["EndTime"] = end.isoformat()
+
+    # Convert the payload dictionary to a JSON string (uses double quotes)
+    payload = json.dumps(payload)
+    # Make data request
+    request = requests.post(url+'/WeatherData', data=payload, headers=headers)
+    # Raise error if request is not OK
+    if request.ok is False:
+        raise ValueError(request.json()['Message'])
+    # Retrieve weather request ID
+    weather_request_id = request.json()["WeatherRequestId"]
+
+    # The SolarAnywhere API is asynchronous, hence a second request is
+    # necessary to retrieve the data (WeatherDataResult).
+    start_time = time.time()  # Current time in seconds since the Epoch
+    # Attempt to retrieve results until the max response time has been exceeded
+    while True:
+        results = requests.get(url+'/WeatherDataResult/'+weather_request_id, headers=headers)  # noqa: E501
+        results_json = results.json()
+        if results_json.get('Status') == 'Done':
+            if results_json['WeatherDataResults'][0]['Status'] == 'Failure':
+                raise RuntimeError(results_json['WeatherDataResults'][0]['ErrorMessages'][0]['Message'])  # noqa: E501
+            break
+        elif (time.time()-start_time) > timeout:
+            raise TimeoutError('Time exceeded the `timeout`.')
+        time.sleep(5)  # Sleep for 5 seconds before each data retrieval attempt
+
+    # Extract time series data
+    data = pd.DataFrame(results_json['WeatherDataResults'][0]['WeatherDataPeriods']['WeatherDataPeriods'])  # noqa: E501
+    # Set datetime index
+    data.index = pd.to_datetime(data['ObservationTime'])
+    if map_variables:
+        data = data.rename(columns=VARIABLE_MAP)
+
+    # Parse metadata
+    meta = results_json['WeatherDataResults'][0]['WeatherSourceInformation']
+    meta['time_resolution'] = results_json['WeatherDataResults'][0]['WeatherDataPeriods']['TimeResolution_Minutes']  # noqa: E501
+    meta['spatial_resolution'] = spatial_resolution
+    # Rename and convert applicable metadata parameters to floats
+    meta['latitude'] = float(meta.pop('Latitude'))
+    meta['longitude'] = float(meta.pop('Longitude'))
+    meta['altitude'] = float(meta.pop('Elevation_Meters'))
+    return data, meta
+
+
+def read_solaranywhere(filename, map_variables=True, encoding='iso-8859-1'):
+    """
+    Read a SolarAnywhere formatted file into a pandas DataFrame.
+
+    The SolarAnywhere file format and variables are described in [1]_. Note,
+    the SolarAnywhere file format resembles the TMY3 file format but contains
+    additional variables and meatadata.
```

Comment:
```suggestion
    additional variables and metadata.
```

Too bad we can't coin this term for substantial metadata, as opposed to thin data.